### PR TITLE
Implement password authentication for SOCKS5

### PR DESCRIPTION
**Pull request by pgerber**
Password support in accordance with [RFC1929](https://tools.ietf.org/html/rfc1929).

### CODE REVIEW

File: src/v5.rs [hunk 1]
```diff
@@ -118,7 +144,7 @@ impl Socks5Stream {
         let packet = [
             5, // protocol version
             1, // method count
-            0, // no authentication
+            auth.id() // auth method
```
Comment (sfackler) [inline src/v5.rs]: We should always declare that we support the no authentication method.

**Comment by sfackler** [comment]
Thanks!

**Comment by jasondavies** [comment]
Would be great to see a new release with this included.  I only realised it was supported when I looked at the code!  Thanks for this very useful library.

**Comment by sfackler** [comment]
Done!

**Pull request merged**